engine/src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -9,19 +10,44 @@ import cromwell.engine.db.DataAccess.{ExecutionKeyToJobKey, WorkflowExecutionAnd
 import cromwell.engine.db.slick._
 import cromwell.engine.workflow.{BackendCallKey, ExecutionStoreKey}
 import cromwell.webservice.{CallCachingParameters, WorkflowQueryParameters, WorkflowQueryResponse}
+import org.slf4j.LoggerFactory
 import wdl4s.values.WdlValue
 import wdl4s.{CallInputs, _}
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
 
 object DataAccess {
+  lazy val log = LoggerFactory.getLogger(classOf[DataAccess])
+
   val globalDataAccess: DataAccess = new slick.SlickDataAccess()
   case class ExecutionKeyToJobKey(executionKey: ExecutionDatabaseKey, jobKey: BackendJobKey)
   case class WorkflowExecutionAndAux(execution: WorkflowExecution, aux: WorkflowExecutionAux)
+
+  // TODO: Nothing DataAccess specific in this retry method. Refactor to lenthall or similar and add a tests.
+  private def withRetry[A](f: => Future[A], delay: FiniteDuration, retries: Int)
+                          (shouldRetry: (Throwable) => Boolean)
+                          (implicit actorSystem: ActorSystem): Future[A] = {
+    // TODO: May want to pass a separate ec, or is the actor system's dispatcher ok?
+    implicit val ec = actorSystem.dispatcher
+    f recoverWith {
+      case throwable if shouldRetry(throwable) && retries > 0 =>
+        log.warn(
+          s"""Transient exception detected: ${throwable.getMessage}
+              |Retrying in $delay ($retries retries remaining)""".stripMargin)
```

Comment:
should this pass the throwable to `log.warn` to get the stack trace, or would that be too noisy?